transport/grpc/transport.go

```diff
@@ -95,36 +96,52 @@ func (t *Transport) NewOutbound(peerChooser peer.Chooser, options ...OutboundOpt
 }
 
 // RetainPeer retains the peer.
-func (t *Transport) RetainPeer(peerIdentifier peer.Identifier, peerSubscriber peer.Subscriber) (peer.Peer, error) {
+//
+// Deprecated: use grpcTransport.NewDialer(...grpc.DialOption) to create a
+// peer.Transport instead of using the grpc.Transport as a peer.Transport.
+// You will need to explicitly choose either grpc.WithInsecure() or
+// grpc.WithCredentials(...) options.
```

Comment:
Outdated comment. Unless I'm misreading the code, we default to WithInsecure if they don't provide credentials.